src/pages/bookings/index.tsx

```diff
@@ -0,0 +1,151 @@
+
+function a11yProps(index: number) {
+  return {
+    'id': `simple-tab-${index}`,
+    'aria-controls': `simple-tabpanel-${index}`,
+  };
+}
+
+export default function Bookings() {
+  const [open, setOpen] = useState(false);
+  const [details, setDetails] = useState<DetailsProps>({
+    id: '',
+    title: '',
+    price: 0,
+    accommodationId: '',
+    thumbnailUrl: '',
+    previewImgUrl: '',
+    startDate: '',
+    endDate: '',
+    status: 'PENDING',
+  });
+  const [value, setValue] = useState<STATUS>('PENDING');
+
+  const { data, isPending, isError, hasNextPage, fetchNextPage } = useGetBookingList(value);
+
+  const bookings = useMemo(
+    () => data?.pages.reduce((acc, page) => [...acc, ...page.data], [] as BookType[]),
+    [data]
+  );
+
+  const handleNextPage = useCallback(() => fetchNextPage(), [fetchNextPage]);
+
+  const handleClick = useCallback((details: DetailsProps) => {
+    setOpen(true);
+    setDetails(details);
+  }, []);
+
+  const handleClose = useCallback(() => setOpen(false), []);
+
+  const handleTabChange = (event: React.SyntheticEvent, newValue: STATUS) => {
+    event.preventDefault();
+    setValue(newValue);
+  };
+
+  const renderAccommodationSkeleton = useCallback(
+    () => (
+      <Box display="grid" gap={8} gridTemplateColumns={'repeat(auto-fill, minmax(280px, 1fr))'}>
+        <AccommodationSkeleton />;
+      </Box>
+    ),
+    []
+  );
+
+  if (isPending) {
+    renderAccommodationSkeleton();
+  }
+  if (isError) {
+    return <DataFetchError errorKey={ErrorTypes.accommodation_failed_to_get_list} />;
+  }
+
+  return (
+    <Box>
+      <Box display="flex" alignItems="center" justifyContent="space-between" my={4}>
+        <Typography variant={'lg'} fontWeight={600}>
+          Your Bookings
+        </Typography>
+      </Box>
+
+      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
+        <Tabs value={value} onChange={handleTabChange} aria-label="basic tabs example">
+          <Tab label="Pending" value="PENDING" {...a11yProps(0)} />
+          <Tab label="Active" value="ACTIVE" {...a11yProps(1)} />
+          <Tab label="Completed" value="COMPLETED" {...a11yProps(2)} />
+        </Tabs>
+      </Box>
+      <Box pt={2}>
+        {bookings?.length === 0 ? (
+          <Typography textAlign={'center'} variant={'h6'} mt={8}>
+            You haven&apos;t booked any accommodations yet
+          </Typography>
+        ) : (
+          <InfiniteScroll
+            next={handleNextPage}
+            hasMore={hasNextPage}
+            dataLength={bookings?.length || 0}
+            loader={renderAccommodationSkeleton()}
+          >
+            <Box
+              display="grid"
+              gap={8}
+              gridTemplateColumns={'repeat(auto-fill, minmax(280px, 1fr))'}
+            >
+              {bookings?.map(
+                ({ id, accommodation, accommodationId, startDate, endDate, status }) => (
+                  <Box
```

Comment:
for better performance create particular memoized list component and drill inside there separately id, accommodation, accommodationId, startDate, endDate, status and onClick handler